frontend/src/vault/keysign/shared/KeysignTxOverview.tsx

```diff
@@ -1,26 +1,58 @@
+import { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import { BrowserOpenURL } from '../../../../wailsjs/runtime/runtime';
 import { useCurrentTxHash } from '../../../chain/state/currentTxHash';
 import { useCopyTxHash } from '../../../chain/ui/hooks/useCopyTxHash';
+import { fromChainAmount } from '../../../chain/utils/fromChainAmount';
 import { getBlockExplorerUrl } from '../../../chain/utils/getBlockExplorerUrl';
+import { useCoinPriceQuery } from '../../../coin/query/useCoinPriceQuery';
+import { useGlobalCurrency } from '../../../lib/hooks/useGlobalCurrency';
 import { IconButton } from '../../../lib/ui/buttons/IconButton';
 import { CopyIcon } from '../../../lib/ui/icons/CopyIcon';
 import { LinkIcon } from '../../../lib/ui/icons/LinkIcon';
 import { HStack, VStack } from '../../../lib/ui/layout/Stack';
+import { MatchQuery } from '../../../lib/ui/query/components/MatchQuery';
 import { Text } from '../../../lib/ui/text';
 import { shouldBePresent } from '../../../lib/utils/assert/shouldBePresent';
+import { formatAmount } from '../../../lib/utils/formatAmount';
 import { Chain } from '../../../model/chain';
+import { CoinMeta } from '../../../model/coin-meta';
+import { KeysignTxOverviewRow } from './KeysignTxOverviewRow';
 import { useKeysignPayload } from './state/keysignPayload';
+import { extractAndFormatFees } from './utils/extractAndFormatFees';
 
 export const KeysignTxOverview = () => {
   const txHash = useCurrentTxHash();
 
   const { t } = useTranslation();
 
   const copyTxHash = useCopyTxHash();
+  const { globalCurrency } = useGlobalCurrency();
+  const {
+    coin: potentialCoin,
+    toAddress,
+    memo,
+    toAmount,
+    blockchainSpecific,
+  } = useKeysignPayload();
+
+  const coin = shouldBePresent(potentialCoin);
+
+  const { decimals } = shouldBePresent(coin);
+  const coinPriceQuery = useCoinPriceQuery(CoinMeta.fromCoin(coin));
+
+  const formattedToAmount = useMemo(() => {
+    if (!toAmount || !coin) return null;
```

Comment:
Also there is no need to do !coin since you already "asserted" it with `shouldBePresent`